autophone.py

```diff
@@ -564,19 +516,42 @@ def read_tests(self):
                         test_devices_repos[device] = t[device].split()
                     configs = config.split()
                     for config_file in configs:
-                        config_file = os.path.join(t['here'], config_file)
-                        tests.append((member_value, config_file,
-                                      test_devices_repos))
+                        config_file = os.path.normpath(
+                            os.path.join(t['here'], config_file))
+                        tests.append((member_value,
+                                      config_file, test_devices_repos))
 
             self._tests.extend(tests)
 
 
     def trigger_jobs(self, data):
         self.logger.info('Received user-specified job: %s' % data)
-        job_data = json.loads(data)
-        if 'build' not in job_data:
+        trigger_data = json.loads(data)
+        if 'build' not in trigger_data:
             return 'invalid args'
-        self.new_job(job_data)
+        build_url = trigger_data['build']
+        tests = []
+        test_names = trigger_data['test_names']
+        if not test_names:
+            # No test names specified, force PhoneTest.match
+            # to return tests with any name.
+            test_names = [None]
+        devices = trigger_data['devices']
+        if not devices:
+            # No devices specified, force PhoneTest.match
+            # to return tests for any device.
+            devices = [None]
+        for test_name in test_names:
+            for device in devices:
```

Comment:
This is getting complicated. We're getting more and more conditions and nested loops...how do you keep it all straight!?

I don't think there's anything actionable here; I'm just feeling concerned by the complexity. Are we asking too much of autophone?